firefox-ios/firefox-ios-tests/Tests/XCUITests/HistoryTests.swift

```diff
@@ -530,9 +530,6 @@ class HistoryTests: BaseTestCase {
     // https://mozilla.testrail.io/index.php?/cases/view/2306890
     // Smoketest
     func testDeleteHistoryEntryBySwiping() {
-        navigateToPage()
-        navigator.goto(LibraryPanel_History)
-        waitForExistence(app.cells.staticTexts["http://example.com/"])
```

Comment:
@clarmso I'm not really sure what's going on here, but this test visits 2 websites, but only deletes 1, and then expects the empty message to show.

Some of my work on this branch fixes the equality check for website duplicates. My guess is that both sites were being deleted before, which was why this test passed?? 🤔

Reply:
I just tried running this test on main and even though the test visits 2 websites, only 1 appears in the history, so I must've fixed another bug in this PR!

Here is the main run:
https://github.com/user-attachments/assets/5ded0f66-6e87-4e0a-84c2-5c041dc8ae51

Reply:
P.S. Here is a video of the run with my fixes on this branch:

https://github.com/user-attachments/assets/11169972-46cc-4bf1-97d2-2405117767aa

Reply:
The deleted lines looks spurious to me after checking with TestRail. :) LGTM.